### ISSUE TITLE
Step in RangeExpression

### ISSUE THREAD

**Issue by Arithmeticus** [qt4cg/qtspecs#695]
In the XPath specs, it seems that a simple modification from...

[34] | RangeExpr | ::= | AdditiveExpr ( "to" AdditiveExpr )?
-- | -- | -- | --

...to... 

[34] | RangeExpr | ::= | AdditiveExpr ( "to" AdditiveExpr ("step" AdditiveExpr)? )?
-- | -- | -- | --

...would be nonintrusive, and bring some nice benefits customary in other PLs, allowing expressions such as `1 to 9 step 2` and `100 to -100 step -4`.

Thoughts?

**Comment by ChristianGruen** [comment]
Such an extension has been discussed on the mailing list a while ago:

https://lists.w3.org/Archives/Public/public-xslt-40/2020Nov/thread.html#msg60
https://lists.w3.org/Archives/Public/public-xslt-40/2020Dec/thread.html#msg12

The tendency was to drop it. One reason was that it's challenging to deal with negative steps without making the solution backward-incompatible (one would need to write `-100 to 100 step -4` instead). 

Alternatives to a `step` or `by` keyword were mentioned in the discussion; maybe those could be added as examples to the spec?

```
reverse(-100 to 100)
(1 to 10)[. mod 2 = 1]
(1 to 5) ! (. * 2)
```

**Comment by michaelhkay** [comment]
Indeed, this was specified (and implemented in Saxon) at one stage.

My recollection is that we decided that the main use cases could be achieved using the `fn:slice()` function, and that delivering new capability through new functions was better than delivering it through new syntax.

**Comment by Arithmeticus** [comment]
Thanks. It's worth adding a note with an xref to `fn:slice` at 4.8.2, no?

**Comment by ChristianGruen** [comment]
@Arithmeticus Sounds good. I took the liberty of assigning the issue to you; I hope that's ok?

I also started assigning other issues to persons (mostly myself) when it seemed more or less clear whose turn it was next.

**Comment by Arithmeticus** [comment]
Yes, thanks for the assignment.

**Comment by Arithmeticus** [comment]
closed with #786
